### PR TITLE
Add a check for namespace existence in kpt live apply

### PR DESCRIPTION
**Pull request by OisinJohnston2005**
Fixes issue #2123, by adding a check for the existence of the namespace in kpt live apply

### CODE REVIEW

File: commands/live/apply/cmdapply_test.go [hunk 1]
```diff
@@ -148,6 +148,7 @@ func TestCmd(t *testing.T) {
 
 			runner := NewRunner(fake.CtxWithDefaultPrinter(), tf, ioStreams, false)
 			runner.Command.SetArgs(tc.args)
+			runner.namespaceChecker = func(_ string) error { return nil }
```
Comment (Catalin-Stratulat-Ericsson) [inline commands/live/apply/cmdapply_test.go:195]: good stuff. correct me if im wrong though but i belive this only checks the "happy path" side of things aka if namespace is given as intended. but not if it goes wrong aka given a invalid ns or not given a ns at all path and check that the error returned is what we expect?

File: commands/live/apply/cmdapply.go [hunk 2]
```diff
@@ -191,6 +196,12 @@ func (r *Runner) runE(c *cobra.Command, args []string) error {
 		return err
 	}
 
+	if !live.NamespaceInObjects(objs, inv.Namespace) {
```
Comment (Catalin-Stratulat-Ericsson) [inline commands/live/apply/cmdapply.go:204]: in the function should we assume the default that if no ns is given we check the default ns the same way as kubectl does? or do we default to invalid return. just a proposition id suggest defaulting same as k8s but just a thought

**Review by Catalin-Stratulat-Ericsson — COMMENTED** [review]
looks good just 2 minor comments

File: commands/live/apply/cmdapply_test.go (hunk 1 above)
Comment (OisinJohnston2005) [reply commands/live/apply/cmdapply_test.go:195]: I wasn't sure how to go about mocking the kubernetes client for the test framework, without this line all of these tests fail, I can do a bit of digging today and see what options are available.

the applyRunner func seems to be overwritten in a similar way one line below this.

File: commands/live/apply/cmdapply_test.go
Comment (Catalin-Stratulat-Ericsson) [inline commands/live/apply/cmdapply_test.go:174]: related to my comment about the "happy path" test cases comment. could we not just add 2/3 more tests here one to check valid ns given, one to check invalid and one to maybe check null value given and one to check our expectation. i haven't fully read through this test framework though so i probably am missing something.

File: commands/live/apply/cmdapply.go (hunk 2 above)
Comment (OisinJohnston2005) [reply commands/live/apply/cmdapply.go:204]: I'll implement defaulting to "default" if inv.Namespace is "", that was an oversight on my part

**Review by mozesl-nokia — DISMISSED** [review]

**Review by mozesl-nokia — APPROVED** [review]

**Review by Catalin-Stratulat-Ericsson — APPROVED** [review]
looks good 👍
